### DISCUSSION TITLE
Auto-shutoff after X seconds of inactivity

### DISCUSSION THREAD

**Discussion by xjdeng** [oobabooga/text-generation-webui#2864] [Ideas]
Would it be possible to implement something that'll shut off the server after X seconds of inactivity?  This would be a godsend for those of us running this on paid Colab or other cloud services when we're, say, training a model or running an API server for some other task and it's finished.  You could also have another parameter like auto-shutoff which terminates the server after X seconds regardless of activity.

Hypothetically, how this would work:

```
!python server.py --inactivity-shutoff 1000 --auto-shutoff 10000
from google.colab import runtime
runtime.unassign()
```

**Comment by Dakota0001** [comment]
I agree, this would be a very useful feature for resource saving.
